hash.go

```diff
@@ -41,8 +41,29 @@ func (s *solver) HashInputs() []byte {
 	buf.WriteString(appenginePkgs)
 
 	// Write each of the packages, or the errors that were found for a
-	// particular subpath, into the hash.
+	// particular subpath, into the hash. We need to do this in a
+	// deterministic order, so expand and sort the map.
+	var sortedPkgs []PackageOrErr
 	for _, perr := range s.rpt.Packages {
+		sortedPkgs = append(sortedPkgs, perr)
+	}
+	sort.Slice(sortedPkgs, func(i, j int) bool {
```

Comment:
is this new? it'd have to be variadic, wouldn't it? (or use reflection, which, ew)